Fix: Failsafe tab not loading

NightHawk32 opened the pull request:
In strict mode functions (toggleStage2) can only be declared immediately within another function. This prevented the failsafe tab from getting loaded properly.

KiteAnton commented:
Is this something that triggers any warning or unwanted behavior? Can't seem to reproduce this issue and the failsafe stage2 hide/show seem to work as expected ?

NightHawk32 commented:
For me Chrome reports errors when clicking on the Failsafe tab and it does not load the tab at all.
![failsafe](https://cloud.githubusercontent.com/assets/12000065/17551912/066bb4ea-5efd-11e6-9834-70fa4e0b30bc.JPG)
![failsafe_1](https://cloud.githubusercontent.com/assets/12000065/17552146/66ec4522-5efe-11e6-9fc9-36ac0d45d724.png)
This happens because the object for the failsafe tab does not get created. For the definition of toggleStage2 Chrome reports the following: SyntaxError: in strict mode code, functions may be declared only at top level or immediately within another function.
After moving the function to the top everything works fine again.

KiteAnton commented:
Was this with latest configurator from master? I don't get the same problem, however I don't see any problem with your change so I think it can be safely merged. Thanks for fixing this.

NightHawk32 commented:
Yes, was the latest version. Maybe this could also be caused by different versions of Chrome. Anyways thanks for merging.

mikeller commented:
This is utterly weird:

- the configurator uses closures all the time, in lots of other places, but only this particular one seems to be blowing up for you;
- I could not find any reference to a restriction on the placement of function definitions in the spec for `use strict;` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode).

What version of Chrome are you using?

NightHawk32 commented:
@mikeller yes, you are right, really strange.
Sorry can't tell you which version it was exactly, since I updated it yesterday, but I have in mind something very old (35.X). But the most probable cause is a bug in Chrome. As soon as I updated Chrome to the latest version (52) I couldn't reproduce this issue any more.
